src/codecovio.jl

```diff
@@ -40,53 +40,138 @@ module Codecov
         return Dict("coverage" => cov)
     end
 
+
+    """
+    kwargs provides default values to insert into args_array, only if they are 
+    not already specified in args_array.
+    """
+    function set_defaults(args_array; kwargs...)
+        defined_names = [k for (k,v) in args_array]
+        for kwarg in kwargs
+            if !(kwarg[1] in defined_names)
+                push!(args_array, kwarg)
+            end
+        end
+      return args_array
+    end
+
+
     """
         submit(fcs::Vector{FileCoverage})
 
-    Take a vector of file coverage results (produced by `process_folder`),
+    Takes a vector of file coverage results (produced by `process_folder`),
     and submits them to Codecov.io. Assumes that this code is being run
-    on TravisCI. If running locally, use `submit_token`.
+    on TravisCI. If running locally, use `submit_local`.
     """
-    function submit(fcs::Vector{FileCoverage})
-        branch  = ENV["TRAVIS_BRANCH"]
-        pr      = ENV["TRAVIS_PULL_REQUEST"]
-        job     = ENV["TRAVIS_JOB_ID"]
-        slug    = ENV["TRAVIS_REPO_SLUG"]
-        build   = ENV["TRAVIS_JOB_NUMBER"]
-        commit  = ENV["TRAVIS_COMMIT"]
-        uri_str = "https://codecov.io/upload/v2?service=travis-org" *
-                    "&branch=$(branch)&commit=$(commit)" *
-                    "&build=$(build)&pull_request=$(pr)&" *
-                    "job=$(job)&slug=$(slug)"
-        println("Codecov.io API URL:")
-        println(uri_str)
+    function submit(fcs::Vector{FileCoverage}; kwargs...)
+        kwargs = set_defaults(kwargs, 
+            service      = "travis-org",
+            branch       = ENV["TRAVIS_BRANCH"],
+            commit       = ENV["TRAVIS_COMMIT"],
+            pull_request = ENV["TRAVIS_PULL_REQUEST"],
+            job          = ENV["TRAVIS_JOB_ID"],
+            slug         = ENV["TRAVIS_REPO_SLUG"],
+            build        = ENV["TRAVIS_JOB_NUMBER"],
+            )
 
-        heads   = Dict("Content-Type" => "application/json")
-        data    = to_json(fcs)
-        req     = Requests.post(URI(uri_str); json = data, headers = heads)
-        println("Result of submission:")
-        dump(req.data)
+        submit_generic(fcs; kwargs...)
     end
 
+
+    """
+        submit_token(fcs::Vector{FileCoverage})
+
+    Takes a `Vector` of file coverage results (produced by `process_folder`),
+    and submits them to Codecov.io. Assumes the submission is being made from 
+    a local git installation.  A repository token should be specified by a 
+    'token' keyword argument or the CODECOV_TOKEN environment variable.
+    """
+    function submit_token(fcs::Vector{FileCoverage}; kwargs...)
+        println("submit_token is deprecated, use submit_local instead")
+        submit_local(fcs; kwargs...)
+    end
+
+    @deprecate submit_token submit_local
```

Comment:
using the `@deprecate` macro should allow you to leave out the placeholder definition of `submit_token` above